sql/core/src/test/scala/org/apache/spark/sql/DataFrameFunctionsSuite.scala

```diff
@@ -735,6 +735,60 @@ class DataFrameFunctionsSuite extends QueryTest with SharedSQLContext {
       df.selectExpr("array_contains(array(1, null), array(1, null)[0])"),
       Seq(Row(true), Row(true))
     )
+
+    checkAnswer(
+      df.selectExpr("array_contains(array(1), 1.23D)"),
+      Seq(Row(false), Row(false))
+    )
+
+    checkAnswer(
+      df.selectExpr("array_contains(array(1), 1.0D)"),
+      Seq(Row(true), Row(true))
+    )
+
+    checkAnswer(
+      df.selectExpr("array_contains(array(1.0D), 1)"),
+      Seq(Row(true), Row(true))
+    )
+
+    checkAnswer(
+      df.selectExpr("array_contains(array(1.23D), 1)"),
+      Seq(Row(false), Row(false))
+    )
+
+    checkAnswer(
+      df.selectExpr("array_contains(array(array(1)), array(1.0D))"),
+      Seq(Row(true), Row(true))
+    )
+
+    checkAnswer(
+      df.selectExpr("array_contains(array(array(1)), array(1.23D))"),
+      Seq(Row(false), Row(false))
+    )
+
+    checkAnswer(
+      df.selectExpr("array_contains(array(array(1)), array(1.23))"),
```

Comment:
hmm? shouldn't this fail because of the bug?

Reply:
@cloud-fan Yes. it should :-) I think i had changed this test case to verify the fix to tighestCommonType.. and pushed it by mistake. Sorry about it.